grails-app/services/org/pih/warehouse/product/ProductService.groovy

```diff
@@ -904,11 +909,47 @@ class ProductService {
     }
 
     /**
-     * Find or create a list of categories with the given names as a map keyed on category name. Nulls, duplicates,
-     * and whitespace will be ignored.
+     * A map of Category keyed on the "desensitized" (ie trimmed and lower-cased) category name.
+     *
+     * We desensitize the name because we want to treat names like "NAME" and "name  " as equivalent, which is useful
+     * when importing data since user input errors are common. In other scenarios this equivalency might not be useful
+     * (we might want to treat "name" and "NAME" as different), so make sure to be intentional when using this class.
+     */
+    private class CategoriesByDesensitizedName {
```

Comment:
I opted to make this a class because it allowed for clearer usage of the map. Users of the class don't need to remember to desensitize the data, it happens automatically, all they need to do is provide the category for puts, and category name for gets